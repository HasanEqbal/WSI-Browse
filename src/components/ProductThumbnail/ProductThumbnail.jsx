import React, { useState } from 'react';
import Parser from 'html-react-parser';
import { Link } from 'react-router-dom';
import './ProductThumbnail.scss';

export default function ProductThumbnail({ product }) {
  const {
    name,
    id,
    price,
    priceRange,
  } = product;

  let regularHigh
  let regularLow
  let sellingHigh
  let sellingLow

  if (priceRange) {
    if (priceRange.regular) {
      regularHigh = priceRange.regular.high;
      regularLow = priceRange.regular.low;
    }

    if (priceRange.selling) {
      sellingHigh = priceRange.selling.high;
      sellingLow = priceRange.selling.low;
    }
  }

  let regularPrice
  let sellingPrice

  if (price) {
    regularPrice = price.regular
    if (price.selling) {
      sellingPrice = price.selling
    }
  }

  let image = product.thumbnail.href;

  return (
    <React.Fragment>
      <div className="cell thumbnail">
        <div className="product-card" >
          <div className="product-card-thumbnail">
            <Link className="product-main-image" to={`/shop/${id}`}><img src={image} alt={name} /></Link>
          </div>
          <h6 className="product-card-title">
            <Link className="product-name"to={`/shop/${id}`}>{Parser(name)}</Link></h6>
          <div>
            {regularHigh ?
              <span className="product-card-old-price">
                Original Price: ${regularLow}-${regularHigh}
              </span>
              : regularPrice ?
                <span className="product-card-regular-price">
                  Original Price: ${regularPrice}
                </span> : ""
            }
          </div>
          <div>
            <span className="product-card-sale-price">
              Sale Price: ${sellingHigh
                ? `${sellingLow} - $${sellingHigh}`
                : sellingPrice
              }
            </span>
          </div>
        </div>
      </div>
    </React.Fragment >
  )
}

