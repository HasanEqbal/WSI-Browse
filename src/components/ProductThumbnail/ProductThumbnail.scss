
.product-card {
  background-color: #fefefe;
  padding: 1rem;
  margin-bottom: 1.5rem;
  margin-right: .8rem;
  border: #D9524F;
}

.product-card-thumbnail {
  display: inline;
  position: relative;
  background-color: #f3f3f3;
}

.product-card-thumbnail:hover{
  border: #D9524F;
}
.product-card-title {
  font-family: 'Open sans', sans-serif;
  color: #082a4c;
  font-size: 0.9rem;
  font-weight: 600;
  margin-top: 1rem;
  height: 4rem;
  margin-bottom: 0;
}


.product-card-sale-price {
  color: #D9524F;
  display: inline-block;
  font-size: 0.75rem;
  font-weight: 600;
  margin-top: 0.8rem;
  margin-left: 0.2rem;
  text-align: center;
}

.product-card-old-price {
  color: darken(#082a4c, 30%);
  display: inline-block;
  font-size: 0.80rem;
  margin-left: 0.3rem;
  margin-right: 0.3rem;
  text-decoration: line-through;
  text-align: center;
}

.product-card-colors {
  display: block;
  margin-top: 0.8rem;
}

.product-card-color-option {
  display: inline-block;
  height: 25px;
  width: 25px;
}

.thumbnail{
  flex-basis: 30%;
}

