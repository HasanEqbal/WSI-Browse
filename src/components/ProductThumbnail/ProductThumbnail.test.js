import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router';
import ProductThumbnail from './ProductThumbnail'
const fixture = require('../../testData/testFixture.json');

describe('Component : Product Details', () => {

  it('expect to render Product Details component', () => {
    const product = fixture.groups[0]
    expect(shallow(<ProductThumbnail product={product} />)).toMatchSnapshot();
  })

  it('verify that product name should display', () => {
    const product = fixture.groups[0];
    const wrapper = shallow(<ProductThumbnail product={product} />)
    expect(wrapper.find('h3')) === fixture.groups[0].name
  })

  it('verify that product image should display', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      priceRange: {
        regular: {
          high: 400,
          low: 40
        }
      }

    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('img').prop("src")) === productdetails.thumbnail.href
  })

  it('verify that original product price should display', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      priceRange: {
        regular: {
          high: 400,
          low: 40
        }
      }
    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('product-card-old-price')) === `Original Price: ${productdetails.priceRange.regular.low}-${productdetails.priceRange.regular.high}`
  })

  it('verify that sale product price should display', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      priceRange: {
        selling: {
          high: 400,
          low: 40
        }
      }

    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('product-card-sale-price')) === `Sale Price: ${productdetails.priceRange.selling.low}-${productdetails.priceRange.selling.high}`
  })

  it('verify that regular product price should display', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      price: {
        regularPrice: 400
      }
    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('product-card-regular-price')) === `Original Price: ${productdetails.price.regularPrice}`
  })

  it('verify that selling product price should display', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      price: {
        selling: 400
      }
    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('product-card-sale-price')) === `Sale Price: ${productdetails.price.selling}`
  })

  it('verify that clicking on the image user should navigate to Product detail page', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      price: {
        selling: 400
      }
    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('.product-main-image').props().to).toEqual('/shop/belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024')
  })

  it('verify that clicking on product name user should navigate to Product detail page', () => {
    const productdetails = {
      name: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      id: "belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024",
      thumbnail: {
        href: "https://www.westelm.com/weimgs/rk/images/wcm/products/202003/0005/belgian-linen-ladder-stripe-embroidery-duvet-cover-shams-s-m.jpg"
      },
      price: {
        selling: 400
      }
    }
    const wrapper = shallow(<ProductThumbnail product={productdetails} />)
    expect(wrapper.find('.product-name').props().to).toEqual('/shop/belgian-linen-embroidery-duvet-cover-shams-slate-white-b3024')
  })


})
