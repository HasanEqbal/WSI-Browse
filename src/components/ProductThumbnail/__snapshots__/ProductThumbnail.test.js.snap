// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : Product Details expect to render Product Details component 1`] = `ShallowWrapper {}`;
